specification/monitor/resource-manager/Microsoft.Insights/stable/2023-12-01/scheduledQueryRule_API.json

```diff
@@ -0,0 +1,784 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to update a scheduled query rule",
+            "schema": {
+              "$ref": "#/definitions/ScheduledQueryRuleResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../common-types/v2/commonMonitoringTypes.json#/definitions/ErrorContract"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update a scheduled query rule": {
+            "$ref": "./examples/patchScheduledQueryRule.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ScheduledQueryRules"
+        ],
+        "description": "Deletes a scheduled query rule.",
+        "operationId": "ScheduledQueryRules_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../common-types/v2/commonMonitoringTypes.json#/definitions/ErrorContract"
+            }
+          },
+          "200": {
+            "description": "Successful request to delete a scheduled query rule"
+          },
+          "204": {
+            "description": "No content: the request was successful, but the response is empty"
+          }
+        },
+        "x-ms-examples": {
+          "Delete a scheduled query rule": {
+            "$ref": "./examples/deleteScheduledQueryRule.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Actions": {
+      "type": "object",
+      "description": "Actions to invoke when the alert fires.",
+      "properties": {
+        "actionGroups": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Action Group resource Ids to invoke when the alert fires."
+        },
+        "customProperties": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
+            "description": "The dictionary of custom properties to include with the post operation. These data are appended to the alert payload."
+          },
+          "description": "The properties of an alert payload."
+        },
+        "actionProperties": {
```

Comment:
Since actionProperties is being added to "stable" API version first time, that's why this linter error is asking about not to use additionalProperties.

A few questions:
1. I don't see any examples getting updated which show the usage of "actionProperties". Please update the examples to show its usage.
2. Why is there a need to a dictionary here? Why can't it be a list of properties, where each object in the array list can be "Property Name" and "Property Value"? That way, you can validate the property name if it is valid or not.
3. How is this different from customProperties?

Reply:
1. Done.
2. This section in the payload is populated with key-value pairs which are passed as-is without validation on our part, via our service's data plane, and are eventually interpreted by a service several hops down the pipe. We do not control the allowed values and we should not validate their content.
3. customProperties are similarly passed as-is, but are strictly for the customer's use. They are added to the notification payload of the alert and are there for the user to categorize and enrich their alerts. They carry no meaning to any service, and therefore are separated from actionProperties.

We explained our scenario in an API Modeling Work Hours session with @rkmanda and he gave us approval for a suppression for the actionProperties field.